### ISSUE TITLE
MailingService Attachments

### ISSUE THREAD

**Issue by shakazulu89** [volkanceylan/Serenity#4377]
Hi
Maybe im missing something simple here but it seems MailingService class doesnt handle attachments at all.
That is, i send an email using MailingService, with attachments. The email gets sent but without the attachments. When i look at that class i dont see any mention of attachments, no attachments in MailRow.cs. 
Whats the dealio?
Thanks

**Comment by edwardch** [comment]
I was testing it out and did something like this (about 10 months ago, haven't looked at it since, but it did work.)

```cs
var doc = connection.List<Documents.Entities.DocumentRow>(d => d
    .Select(Documents.Entities.DocumentRow.Fields.Path)
    .Select(Documents.Entities.DocumentRow.Fields.OriginalName)
    .From("Documents")
    .Where(Documents.Entities.DocumentRow.Fields.DocumentId < 10) // for testing
).FirstOrDefault();

if (doc != null)
{
    var pathInfo = doc.Path;
    var attachmentPath = Serenity.Web.UploadHelper.DbFilePath(pathInfo);
    var attachmentName = doc.OriginalName;
    new MailingService().SendById(connection, mail.MailId.Value, attachmentPath, attachmentName);
} else
{
    new MailingService().SendById(connection, mail.MailId.Value, "", "");
}
```

**Comment by shakazulu89** [comment]
@edwardch  thanks for the reply and sorry for the long delay!
where do you get "mail.MailId.Value" from?

**Comment by edwardch** [comment]
MailId is in MailRow, and the "mail" object came from the database in a connection.List<MailRow> call which you should have in your code somewhere. I think this is part of StartSharp (assuming you have access because your question mentions Mail Row/Service) so I don't think I can share full code here...

**Comment by shakazulu89** [comment]
i do have startsharp. but the project was built before i had startsharp and i havent updated it yet.
so you first create and insert a mail row, then list all pending mails then invoke sendbyid?
but i still dont really understand. sendbyid has no declaration for attachments. neither does BuildMessage and neither MailRow.
i can modify these files to handle attachments but i just wanted to be sure im not missing something thats already there.

**Comment by edwardch** [comment]
> so you first create and insert a mail row, then list all pending mails then invoke sendbyid?

Yep, in a background task...

> i can modify these files to handle attachments but i just wanted to be sure im not missing something thats already there.

Yes sorry I should've specified that I modified both of these methods to handle attachments.

This was purely something that I was just testing out so it might be better to have an attachment list on the MailRow. The example I gave was literally getting the first document from my "Documents" table and sending that as the attachment!
